playbooks/collect-logs.yml

```diff
@@ -114,6 +118,36 @@
         segfault_problems_found: "{{ result.stdout_lines }}"
       ignore_errors: yes
 
+    - name: Collect logs from nova
+      sudo: yes
+      shell: cp -Rf /var/log/nova/* /var/log/extra/openstack
+      ignore_errors: yes
+
+    - name: Collect logs from glance
+      sudo: yes
+      shell: cp -Rf /var/log/glance/* /var/log/extra/openstack
+      ignore_errors: yes
+
+    - name: Collect logs from cinder
+      sudo: yes
+      shell: cp -Rf /var/log/cinder/* /var/log/extra/openstack
+      ignore_errors: yes
+
+    - name: Collect logs from keystone
+      sudo: yes
+      shell: cp -Rf /var/log/keystone/* /var/log/extra/openstack
+      ignore_errors: yes
+
+    - name: Collect logs from neutron
+      sudo: yes
+      shell: cp -Rf /var/log/neutron/* /var/log/extra/openstack
+      ignore_errors: yes
+
+    - name: Collect logs from horizon
+      sudo: yes
+      shell: cp -Rf /var/log/httpd/horizon_error.log /var/log/extra/openstack
+      ignore_errors: yes
+
```

Comment:
with_items will run over all the items even if someone will fail. If you put ignore_errors: yes it will ignore any errors. I've tested that with simple playbook: 
```
- hosts: localhost
  tasks:
    - file:
        path: "{{item}}"
        state: file
      ignore_errors: yes
      with_items:
        - /bin/true
        - "I am here"
        - /bin/false
```